### ISSUE TITLE
func-name-matching: New option for property descriptors to match against property name

### ISSUE THREAD

**Issue by platinumazure** [eslint/eslint#9926]
**What rule do you want to change?**

func-name-matching

**Does this change cause the rule to produce more or fewer warnings?**

Sometimes more, sometimes fewer. See examples.

**How will the change be implemented? (New option, new default behavior, etc.)?**

New options: `considerPropertyDescriptor: true` (false by default).

**Please provide some example code that this change will affect:**

```js
Object.defineProperty(model, 'key', {
  value: function key() {
    //
  }
});

Object.defineProperties(model, {
  key: {
    value: function key() {
      //
    }
  }
});

Object.create(Foo.prototype, {
  key: {
    value: function key() {
      //
    }
  }
});
```

**What does the rule currently do for this code?**

All three have errors: "Function `key` should match property name 'value'."

**What will the rule do after it's changed?**

If `{ considerPropertyDescriptor: true }` is provided in configuration, all 3 examples should pass.

However, examples like these should then start to be warned:

```js
Object.defineProperty(model, 'key', {
  value: function value() {  // Function name `value` should match property name `key`.
    //
  }
});

Object.defineProperties(model, {
  key: {
    value: function value() {  // Function name `value` should match property name `key`.
      //
    }
  }
});

Object.create(Foo.prototype, {
  key: {
    value: function value() {  // Function name `value` should match property name `key`.
      //
    }
  }
});
```

-------

This is another attempt to propose the same thing as #7423. Unfortunately, there were some process hiccups and the contributor no longer seems interested. So I want to carry it forward.

I'll champion.

**Comment by platinumazure** [comment]
@eslint/eslint-team Any thoughts on this proposal? Just asking before I close in a few days. Thanks!

**Comment by btmills** [comment]
Just added my :+1:

**Comment by kaicataldo** [comment]
This has now been accepted.
